src/crxviewer.js

```diff
@@ -887,6 +895,73 @@ var viewFileInfo = (function() {
     return viewFileInfo;
 })();
 
+function parseExtensionFromManifest(manifest) {
+    var id;
+    if (manifest.browser_specific_settings &&
+        manifest.browser_specific_settings.gecko &&
+        (id = manifest.browser_specific_settings.gecko.id)) {
+        return id;
+    }
+
+    if (manifest.applications && manifest.applications.gecko &&
+        (id = manifest.applications.gecko.id)) {
+        return id;
+    }
+
+    return '';
+}
+
+// Extract the extension ID from the currently loaded extension. The ID will be
+// determined from the COSE signature, the PKCS#7 signature, or manifest.json,
+// whatever comes first. Resolves to the extension ID (and the extension type
+// between parentheses if available), or rejects on parser errors.
+function calculateExtensionID() {
+    var fileList = document.getElementById('file-list');
+    var readEntry = function(filename, Writer) {
+        var li = fileList.querySelector('li[data-filename="' + filename + '"]');
+        if (!li) {
+            return null;
+        }
+        return new Promise(function(resolve) {
+            li.zipEntry.getData(new Writer(), function(data) {
+                resolve(data);
+            });
+        });
+    };
+    var formatSubject = function(subject) {
+        return subject.CN + ' (' + subject.OU + ')';
```

Comment:
Let's translate OU:

```
Mozilla Components = system
Mozilla Extensions = privileged
else = signed
```

If you want to, you can use `console.log` to print the subject and/or OU for more verbosity.

Reply:
Should I parse O or OU? For the leaf certificate, we have `/O=Addons/OU=Production`
The issuer is `/OU=Mozilla AMO Production Signing Service` (I have omitted some other fields for brevity).